src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -93,6 +93,7 @@ MultiClientConnect(const char *nodeName, uint32 nodePort, const char *nodeDataba
 	int32 connectionId = AllocateConnectionId();
 	char *effectiveDatabaseName = NULL;
 	char *effectiveUserName = NULL;
+	int timeoutInSeconds = NodeConnectionTimeout / 1000;
```

Comment:
This will round to zero if timeout is set to less than a sec... We probably want to round up instead of down here? A longer timeout is ok, a shorter one not so much.